### PR TITLE
flash_map: move flash areas redirect labels to one header

### PR DESCRIPTION
**Pull request by nvlsianpu**
Labels for assign proper flash area id which depends on TrustZone enabled or not 
were distributed among the source code.
This patch move them into one header file.

Signed-off-by: Andrzej Puzdrowski andrzej.puzdrowski@nordicsemi.no

### CODE REVIEW

**Review by hakonfam — APPROVED** [review]

**Review by SebastianBoe — APPROVED** [review]

**Review by sigvartmh — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Comment by Laczen** [comment]
All these defines are related to mcuboot, they should be moved to a folder related to mcuboot and not in the main zephyr include. Also the name of the include file should be related to mcuboot.
Ideally the defines for mcuboot should not be generated by the DTS file but by a mcuboot config file.

**Comment by nvlsianpu** [comment]
Storage label can land here as well in near feature. It didn't so fare because it is identical for Secure and NonSecure build right now. My point is that partitioning is not only related to mcuboot, it imposes on storage as well.

**Comment by Laczen** [comment]
@nvlsianpu, where the storage label should land is something else which should also be decided. It has nothing to do with the fact that all these labels are mcuboot related.

Suppose you would integrate the nrf bootloader next to mcuboot, you would need to have other labels defined for this (i.e. for the MBR region). The definitions for the nrf bootloader should then be configured using Kconfig.

Once this approach is followed you would need to decide on storage: is this defined by the bootloader (or is the bootloader putting constraints on storage) than the storage setting needs to be (re)defined by
the bootloader settings. Since storage is a zephyr related setting a default could be defined in the dts so
that compilation without any bootloader would also provide defines for storage.

**Review by galak — CHANGES_REQUESTED** [review]
We're trying to reduce headers in include/ or at least the pollution of it.  While I see the duplication going on here, I'm not sure this belongs in include/, not sure where to put it.

File: samples/drivers/soc_flash_nrf/src/main.c [hunk 1]
```diff
@@ -9,13 +9,9 @@
 #include <flash.h>
 #include <device.h>
 #include <stdio.h>
+#include <flash_areas.h>
 
-
-#ifdef CONFIG_TRUSTED_EXECUTION_NONSECURE
-#define FLASH_TEST_OFFSET DT_FLASH_AREA_IMAGE_1_NONSECURE_OFFSET
```
Comment (nvlsianpu) [inline samples/drivers/soc_flash_nrf/src/main.c]: this is offset, ot id so is not included in the heder added, fix it

**Comment by nvlsianpu** [comment]
^^
@sigvartmh @hakonfam What do you think about the header location?

**Comment by Laczen** [comment]
@andrewboie, @vanwinkeljan, would this not be the ideal moment to start moving the mcuboot defines into a seperate mcuboot config setting instead of using dts ? The present proposal would distribute the mcuboot settings even more throughout the zephyr source.  I remember this being discussed in #15831.

**Comment by nvlsianpu** [comment]
@Laczen This will be probably something we need to do after multi-mage build PR will be adapted to zephyr. Now I would hesitate against touching that,

**Comment by nvlsianpu** [comment]
I will move file to mcuboot related place.

**Comment by Laczen** [comment]
> @Laczen This will be probably something we need to do after multi-mage build PR will be adapted to zephyr. Now I would hesitate against touching that,

Maybe, if the multi-image build is a zephyr related PR it should not use any mcuboot related defines. As far as I understand it there should be 2 (or more) text offset defines, these should not depend on mcuboot generated defines. When mcuboot is selected these text offsets could be redefined to match correct image definitions and locations allowing for a header to be added.

**Comment by nvlsianpu** [comment]
@galak pleas revisit

**Comment by nvlsianpu** [comment]
@galak ping

**Comment by nashif** [comment]
stale, closing. Reopen when there are updates...

**Pull request closed without merging**